src/tests/sqlsmith/src/relation.rs

```diff
@@ -32,28 +32,31 @@ fn create_join_on_clause(left: String, right: String) -> Expr {
 
 impl<'a, R: Rng> SqlGenerator<'a, R> {
     /// A relation specified in the FROM clause.
-    pub(crate) fn gen_from_relation(&mut self) -> TableWithJoins {
-        match self.rng.gen_range(0..=9) {
+    pub(crate) fn gen_from_relation(&mut self) -> (TableWithJoins, Vec<Table>) {
+        let range = if self.can_recurse() { 10 } else { 9 };
```

Comment:
This works for now. Equijoin may also become recursive in future. We can revisit it then. Tracked here: #3896 under refinements.

Reply:
alright, we can change the value whenever we need in future.